go/DEVELOPER.md

```diff
@@ -238,6 +238,132 @@ go run . -help
 go run . -resultsFile gobenchmarks.json -dataSize "100 1000" -concurrentTasks "10 100" -clients all -host localhost -port 6379 -clientCount "1 5" -tls
 ```
 
+### Naming Conventions
+
+#### Function names
+
+For every command that you implement, please use PascalCase.
+
+Examples of good command function names:
+
+- `BZPopMin`
+- `ZRem`
+- `PExpireWithOptions`
+- `SetWithOptions`
+- `Decr`
+
+Examples of bad command function names:
+
+- `zRange`
+- `hincrbyfloat`
+- `Sdiffstore`
+
+#### Function names for examples
+
+`pkgsite` is a program which generates documentation for Go projects.  To install and run it, execute the following:
+
+```bash
+# In the valkey-glide directory
+cd go
+go install golang.org/x/pkgsite/cmd/pkgsite@latest
+pkgsite -open .
+```
+
+In order for `pkgsite` to work properly, examples for API must be written in a very specific format. They should be located in one of the `*_test.go` files in `./api/` and follow this pattern: `Example<ClientType>_<FunctionName>()`. If we wanted to create an example for the `Get` command in `GlideClient`, we would name define our function as follows:
+
+```go
+func ExampleGlideClient_Get() {
+    // Example code here
+}
+```
+
+In cases where we want to show more than one example, we can add extra endings to the function names:
```

Comment:
Endings must begin with a lower case letter.  Examples:
* ExampleGlideClient_Get_KeyExists // bad: extension begins with upper case
* ExampleGlideClient_Get_1 // bad: extension begins with number
* ExampleGlideClient_Get_one // good: extension begins with lower case letter
* ExampleGlideClient_Get_keyIsNil // better: extension begins with lower case letter and is descriptive